### ISSUE TITLE
Issue with async*, error handling and dart2js

### ISSUE THREAD

**Issue by NevRA** [dart-lang/sdk#35624]
I have a code:
`Future<int> doJobFuture(int i) async {
  throw 'error';
}

Stream<int> doJob(Stream<int> input) async* {
  await for (final i in input) {
    try {
      final res = await doJobFuture(i);
      yield res;
    } catch (e) {
      print(e);
    }
  }
}

void main() async {
  final sc = new StreamController<int>.broadcast(sync: false);
  doJob(sc.stream).listen((t) => print(t));

  await Future.delayed(const Duration(seconds: 1), () {
    sc.add(1);
    sc.add(1);
    sc.add(1);
  });
}`

It works fine on VM and I can see 3 errors messages in console.
If I run this example on dart2js (or ddc) I'll get only 1 error with 1 actual attempt to execute doJobFuture method.

Looks like something wrong with exceptions handling from Future inside 'await for' block.

If I change final res = await doJobFuture(i); to the throw Exception('error'); I'll get expected result.
If I change try - catch block to the Future methods (then, cathcError) it works fine too.

Dart VM version: 2.1.0 (Tue Nov 13 18:22:02 2018 +0100) on "macos_x64"
Chrome
MacOS

**Comment by natebosch** [comment]
I'm only able to reproduce with DDC, not with dart2js

**Comment by NevRA** [comment]
@lrhn yes, dartpad uses VM that is why it works fine.
@natebosch strange, I'll double check it.

**Comment by NevRA** [comment]
Yeah, you are right guys. I checked and issue related ONLY to the DDC. Dev experience still may be affected but no issues on prod. Sorry for the confusion.
I replaced dart2js with ddc in description
Thanks!
